android-project/app/src/main/java/oceantreasur/es/view/CustomButton.java

```diff
@@ -0,0 +1,46 @@
+package oceantreasur.es.view;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.graphics.Typeface;
+import android.util.AttributeSet;
+import android.util.Log;
+
+import oceantreasur.es.R;
+
+public class CustomButton extends android.support.v7.widget.AppCompatButton {
+
+    public CustomButton(Context context) {
+        super(context);
+    }
+
+    public CustomButton(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        TypedArray ta = context.obtainStyledAttributes(attrs, R.styleable.CustomButton);
+
+        applyCustomFont(ta, attrs);
+    }
+
+    public CustomButton(Context context, AttributeSet attrs, int defStyle) {
+        super(context, attrs, defStyle);
+        TypedArray ta = context.obtainStyledAttributes(attrs, R.styleable.CustomButton);
+
+        applyCustomFont(ta, attrs);
+    }
+
+    public void applyCustomFont(TypedArray ta, AttributeSet attrs) {
+
+        if (ta != null) {
```

Comment:
Redundant code.
Move this logic in a static method in FontManager. Method should take the concrete custom attribute id (R.styleable.CustomTextView_typefaceAsset), the custom view id (R.styleable.CustomTextView) and the available AttributeSet. Please refer to the CustomTextView constructor as it is more complete.